pilot/pkg/networking/util/util.go

```diff
@@ -257,15 +257,21 @@ func SortVirtualHosts(hosts []route.VirtualHost) {
 	})
 }
 
-// IsProxyVersionGE11 checks whether the given Proxy version is greater than or equals 1.1.
-func IsProxyVersionGE11(node *model.Proxy) bool {
-	ver, _ := node.GetProxyVersion()
-	return ver >= "1.1"
+// IsIstioVersionGE12 checks whether the given Istio version is greater than or equals 1.2.
+func IsIstioVersionGE12(node *model.Proxy) bool {
+	ver, _ := node.GetIstioVersion()
+	return ver >= "1.2"
```

Comment:
I thought you were going to check for `release-1.2`, etc as well? We cannot break all of our daily builds -- real users uses these as well not just devs

Reply:
I realized so those checks are not right.. Its because checking for release-1.2 means we are asking for equality (is this a 1.2 proxy) and not a greater than or equal to! Secondly, this code is not going into release-1.2 branch (we shouldn't be cherry picking this). So code in release-1.2 will continue to work as is. This code is applicable only for release-1.3 branch, where we should try to set versions to 1.3.x or so..

Reply:
yes but now say we release 1.3 with protocol sniffing. A user installs 1.3, and their proxies with version release-1.2 now all get sent this unsupported feature and break. release-1.2 cannot be considered greater than release 1.3